File: src/types/weather.ts
export interface WeatherData {
  stationName: string;
  state: string;
  temperatureC: number;
  feelsLikeC: number;
  condition: string;
  rainfallPast24hMm: number;
  precipitationProbability: number;
  windSpeedKmh: number;
  windGustKmh: number;
  windDirection: string;
  humidityPercent: number;
  uvIndex: number;
  aqiValue: number;
  aqiCategory: "Good" | "Satisfactory" | "Moderate" | "Poor" | "Very Poor" | "Severe";
  radarStatus: "Operational" | "Maintenance" | "Warning Mode";
  radarStation: string;
  imdWarningColor: "Red" | "Orange" | "Yellow" | "Green";
  hourlyForecast: {
    time: string;
    tempC: number;
    rainProb: number;
    condition: string;
  }[];
}

File: backend/app/schemas/weather.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class HourlyForecastItem(BaseModel):
    time: str
    temp_c: float
    rain_prob: float
    condition: str


class CurrentWeatherResponse(BaseModel):
    station_name: str
    state: str
    latitude: float
    longitude: float
    temperature_c: float
    feels_like_c: float
    condition: str
    rainfall_past_24h_mm: float
    precipitation_probability: float
    wind_speed_kmh: float
    wind_gust_kmh: float
    wind_direction: str
    humidity_percent: float
    uv_index: float
    aqi_value: int
    aqi_category: str
    radar_status: str
    radar_station: str
    imd_warning_color: str = Field(..., description="Red, Orange, Yellow, Green")
    hourly_forecast: List[HourlyForecastItem] = Field(default_factory=list)


class WeatherTelemetryRead(BaseModel):
    station_code: str
    station_name: str
    state: str
    temperature_c: float
    wind_speed_kmh: float
    precipitation_mm: float
    imd_warning_color: str
    recorded_at: datetime
